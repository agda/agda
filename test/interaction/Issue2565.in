top_command (cmd_load currentFile [])
-- Need actual ranges here for 'literal give' to kick in.
IOTCM currentFile None Indirect (Cmd_give WithoutForce 0 (intervalsToRange Nothing [Interval () (Pn () 1 2 3) (Pn () 1 2 3)]) "4")
IOTCM currentFile None Indirect (Cmd_give WithoutForce 1 (intervalsToRange Nothing [Interval () (Pn () 1 2 3) (Pn () 1 2 3)]) "fromNat 4")
IOTCM currentFile None Indirect (Cmd_give WithoutForce 2 (intervalsToRange Nothing [Interval () (Pn () 1 2 3) (Pn () 1 2 3)]) "\"bar\"")
IOTCM currentFile None Indirect (Cmd_give WithoutForce 3 (intervalsToRange Nothing [Interval () (Pn () 1 2 3) (Pn () 1 2 3)]) "fromString \"bar\"")
IOTCM currentFile None Indirect (Cmd_give WithoutForce 4 (intervalsToRange Nothing [Interval () (Pn () 1 2 3) (Pn () 1 2 3)]) "-4")
IOTCM currentFile None Indirect (Cmd_give WithoutForce 5 (intervalsToRange Nothing [Interval () (Pn () 1 2 3) (Pn () 1 2 3)]) "fromNeg 4")
goal_command 6 cmd_refine "_+_ 4"
goal_command 7 cmd_refine "_+_ (fromNat 4)"
IOTCM currentFile None Indirect (Cmd_refine 8 (intervalsToRange Nothing [Interval () (Pn () 1 2 3) (Pn () 1 2 3)]) "append \"foo\"")
IOTCM currentFile None Indirect (Cmd_refine 9 (intervalsToRange Nothing [Interval () (Pn () 1 2 3) (Pn () 1 2 3)]) "append (fromString \"foo\")")
goal_command 10 cmd_refine "-4 +Z_"
goal_command 11 cmd_refine "fromNeg 4 +Z_"
