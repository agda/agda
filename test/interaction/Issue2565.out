(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Nat ?1 : Nat ?2 : List Char ?3 : List Char ?4 : Int ?5 : Int ?6 : Nat ?7 : Nat ?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\"" nil)
((last . 1) . (agda2-goals-action '(0 1 2 3 4 5 6 7 8 9 10 11 12)))
(agda2-give-action 0 'no-paren)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?1 : Nat ?2 : List Char ?3 : List Char ?4 : Int ?5 : Int ?6 : Nat ?7 : Nat ?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\"" nil)
((last . 1) . (agda2-goals-action '(1 2 3 4 5 6 7 8 9 10 11 12)))
(agda2-give-action 1 'paren)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?2 : List Char ?3 : List Char ?4 : Int ?5 : Int ?6 : Nat ?7 : Nat ?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\"" nil)
((last . 1) . (agda2-goals-action '(2 3 4 5 6 7 8 9 10 11 12)))
(agda2-give-action 2 'no-paren)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?3 : List Char ?4 : Int ?5 : Int ?6 : Nat ?7 : Nat ?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\"" nil)
((last . 1) . (agda2-goals-action '(3 4 5 6 7 8 9 10 11 12)))
(agda2-give-action 3 'paren)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?4 : Int ?5 : Int ?6 : Nat ?7 : Nat ?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\"" nil)
((last . 1) . (agda2-goals-action '(4 5 6 7 8 9 10 11 12)))
(agda2-give-action 4 'no-paren)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?5 : Int ?6 : Nat ?7 : Nat ?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\"" nil)
((last . 1) . (agda2-goals-action '(5 6 7 8 9 10 11 12)))
(agda2-give-action 5 'paren)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?6 : Nat ?7 : Nat ?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\"" nil)
((last . 1) . (agda2-goals-action '(6 7 8 9 10 11 12)))
(agda2-give-action 6 "4 + ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?7 : Nat ?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\" ?13 : Nat" nil)
((last . 1) . (agda2-goals-action '(13 7 8 9 10 11 12)))
(agda2-give-action 7 "fromNat 4 + ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?8 : List Char ?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\" ?13 : Nat ?14 : Nat" nil)
((last . 1) . (agda2-goals-action '(13 14 8 9 10 11 12)))
(agda2-give-action 8 "append \"foo\" ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?9 : List Char ?10 : Int ?11 : Int ?12 : P 4 -5 \"6\" ?13 : Nat ?14 : Nat ?15 : List Char" nil)
((last . 1) . (agda2-goals-action '(13 14 15 9 10 11 12)))
(agda2-give-action 9 "append (fromString \"foo\") ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?10 : Int ?11 : Int ?12 : P 4 -5 \"6\" ?13 : Nat ?14 : Nat ?15 : List Char ?16 : List Char" nil)
((last . 1) . (agda2-goals-action '(13 14 15 16 10 11 12)))
(agda2-give-action 10 "-4 +Z ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?11 : Int ?12 : P 4 -5 \"6\" ?13 : Nat ?14 : Nat ?15 : List Char ?16 : List Char ?17 : Int" nil)
((last . 1) . (agda2-goals-action '(13 14 15 16 17 11 12)))
(agda2-give-action 11 "fromNeg 4 +Z ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?12 : P 4 -5 \"6\" ?13 : Nat ?14 : Nat ?15 : List Char ?16 : List Char ?17 : Int ?18 : Int" nil)
((last . 1) . (agda2-goals-action '(13 14 15 16 17 18 12)))
